op-batcher/batcher/channel_manager.go

```diff
@@ -106,9 +107,11 @@ func (s *channelManager) TxConfirmed(_id txID, inclusionBlock eth.BlockID) {
 	if channel, ok := s.txChannels[id]; ok {
 		delete(s.txChannels, id)
 		done, blocks := channel.TxConfirmed(id, inclusionBlock)
-		s.blocks = append(blocks, s.blocks...)
 		if done {
 			s.removePendingChannel(channel)
+			if len(blocks) > 0 {
```

Comment:
You could add the length check at the top of `Prepend` and let it be a noop in case the length is 0. Then we don't need to add the check every time we use `Prepend`. Or do you add this here for clarity?

Reply:
Yes I think I was trying to go for clarity here, prepending the blocks seems like quite special case which I wanted to make clear is only being done when `channel.TxConfirmed` signals it.

Reply:
Sounds good, I still think it's worth _also_ adding that check to the `Prepend` implementation, to avoid unnecessarily creating a new slice every time 0 blocks are prepended. I'm not sure if the Go compiler is smart enough to do this automatically.

Reply:
Agreed, I had in fact already done that here https://github.com/ethereum-optimism/optimism/pull/12180/commits/ffc238c658d6c145c7e1109a44fc1e2cf4b5da22